### PR TITLE
CLOUDSTACK-10163: Component tests sanity

### PR DESCRIPTION
**Pull request by borisstoyanov**
I've fixed quite a few component tests, most of them should pass now.

### CODE REVIEW

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan help

**Comment by rhtyd** [comment]
@borisstoyanov can you fix conflicts, let's re-review/test and merge this?

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 kvm-centos7 component/test_*.py

**Comment by rhtyd** [comment]
Rebased and fixed conflicts @borisstoyanov 
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 kvm-centos7 component/test_lb_secondary_ip.py component/test_acl_listvm.py component/test_snapshots.py component/test_ldap_auto_import.py component/test_host_maintenance.py component/test_assign_vm.py component/test_invalid_gw_nm.py component/test_cpu_limits.py component/test_storage_motion.py component/test_host_ha.py component/test_escalations_routers.py component/test_secsr_mount.py component/test_escalations_volumes.py component/test_recurring_snapshots.py component/test_browse_volumes.py component/test_stopped_vm.py component/test_allocation_states.py component/test_snapshot_gc.py component/test_mm_max_limits.py component/test_interop_xd_ccp.py component/test_acl_listvolume.py component/test_tags.py component/test_affinity_groups_projects.py component/test_resource_limits.py component/test_security_groups.py component/test_escalations_ipaddresses.py component/test_cpu_project_limits.py component/test_mm_domain_limits.py component/test_cpu_max_limits.py component/test_ps_max_limits.py component/test_vpc_network_internal_lbrules.py component/test_ps_resize_volume.py component/test_escalations_templates.py component/test_vpc_network.py component/test_eip_elb.py component/test_deploy_vgpu_vm.py component/test_vm_passwdenabled.py component/test_browse_templates2.py component/test_separate_backup_from_snapshot.py component/test_escalations_vpncustomergateways.py component/test_project_resources.py component/test_multiple_ips_per_nic.py component/test_template_from_snapshot_with_template_details.py component/test_add_remove_network.py component/test_ncc_integration_dedicated.py component/test_acl_isolatednetwork_delete.py component/test_templates.py component/test_overcommit.py component/test_reset_ssh_keypair.py component/test_ps_resource_limits_volume.py component/test_escalations_vmware.py component/test_mm_project_limits.py component/test_simultaneous_volume_attach.py component/test_acl_listsnapshot.py component/test_netscaler_nw_off.py component/test_escalations_snapshots.py component/test_volumes.py component/test_regions.py component/test_ss_project_limits.py component/test_routers.py component/test_vpn_users.py component/test_host.py component/test_dhcp_dns_offload.py component/test_escalations_instances.py component/test_redundant_router_cleanups.py component/test_cpu_domain_limits.py component/test_persistent_networks.py component/test_usage.py component/test_ss_max_limits.py component/test_netscaler_lb_algo.py component/test_accounts.py component/test_projects.py component/test_vpc_vm_life_cycle.py component/test_ss_domain_limits.py component/test_haproxy.py component/test_vpc.py component/test_blocker_bugs.py component/test_rootvolume_resize.py component/test_netscaler_lb.py component/test_asa1000v_fw.py component/test_netscaler_lb_sticky.py component/test_organization_states.py component/test_project_usage.py component/test_regions_accounts.py component/test_ncc_integration_shared.py component/test_memory_limits.py component/test_base_image_updation.py component/test_deploy_vm_userdata_reg.py component/test_project_limits.py component/test_baremetal.py component/test_acl_isolatednetwork.py component/test_brocade_vcs.py component/test_bigswitch_bcf.py component/test_concurrent_create_volume_from_snapshot.py component/test_escalations_networks.py component/test_custom_hostname.py component/test_network_offering.py component/test_browse_templates.py component/test_project_configs.py component/test_VirtualRouter_alerts.py component/test_vpc_vms_deployment.py component/test_vpc_offerings.py component/test_acl_sharednetwork_deployVM-impersonation.py component/test_vpn_service.py component/test_implicit_planner.py component/test_deploy_vm_userdata_multi_nic.py component/test_affinity_groups.py component/test_redundant_router_upgrades.py component/test_advancedsg_networks.py component/test_vpc_network_pfrules.py component/test_ldap.py component/test_netscaler_configs.py component/test_update_vm.py component/test_egress_fw_rules.py component/test_multiple_ip_ranges.py component/test_ps_limits.py component/test_ip_reservation.py component/test_ps_domain_limits.py component/test_region_vpc.py component/test_dynamic_compute_offering.py component/test_ss_limits.py component/test_ps_project_limits.py component/test_escalation_listTemplateDomainAdmin.py component/test_multiple_public_interfaces.py component/test_escalations_securitygroups.py component/test_vpc_distributed_routing_offering.py component/test_non_contiguous_vlan.py component/test_shared_networks.py component/test_portable_ip.py component/test_escalations_isos.py component/test_list_nics.py component/test_snapshots_improvement.py component/test_vpc_network_lbrules.py component/test_egress_rules.py component/test_vpc_routers.py component/test_vmware_drs.py component/test_acl_sharednetwork.py component/test_concurrent_snapshots_limit.py component/test_vpc_network_staticnatrule.py component/test_list_pod.py component/test_migrate_vol_to_maintained_pool.py component/test_snapshot_limits.py component/test_redundant_router_services.py

**Comment by rhtyd** [comment]
Pinging for additional review - @nvazquez @DaanHoogland @wido @marcaurele @rafaelweingartner and others, thanks.

**Comment by borisstoyanov** [comment]
Many of these are issues that are not fixed yet, others may be regressions, but the failure number is down to 41 from ~100
ping @PaulAngus @DaanHoogland @nvazquez for review and lgtm.

File: test/integration/component/test_VirtualRouter_alerts.py [hunk 1]
```diff
@@ -47,6 +47,7 @@ def setUpClass(cls):
             cls).getClsTestClient()
         cls.api_client = cls.testClient.getApiClient()
         cls.services = cls.testClient.getParsedTestDataConfig()
+        cls.hostConfig = cls.config.__dict__["zones"][0].__dict__["pods"][0].__dict__["clusters"][0].__dict__["hosts"][0].__dict__
```
Comment (nvazquez) [inline test/integration/component/test_VirtualRouter_alerts.py]: Is `cls.config` set here or should it be `cls.services?` Can we add basic check for each dictionary key?

Comment (borisstoyanov) [reply test/integration/component/test_VirtualRouter_alerts.py]: This is a configuration, not a test data which we consider as 'services'.

File: test/integration/component/test_escalations_routers.py [hunk 2]
```diff
@@ -46,6 +46,7 @@ def setUpClass(cls):
             cls).getClsTestClient()
         cls.api_client = cls.testClient.getApiClient()
         cls.services = cls.testClient.getParsedTestDataConfig()
+        cls.hostConfig = cls.config.__dict__["zones"][0].__dict__["pods"][0].__dict__["clusters"][0].__dict__["hosts"][0].__dict__
```
Comment (nvazquez) [inline test/integration/component/test_escalations_routers.py]: Same question here

File: test/integration/component/test_host_ha.py [hunk 3]
```diff
@@ -45,6 +45,8 @@ def setUp(self):
         self.services = self.testClient.getParsedTestDataConfig()
         self.zone = get_zone(self.apiclient, self.testClient.getZoneForTests())
         self.pod = get_pod(self.apiclient, self.zone.id)
+        self.hostConfig = self.config.__dict__["zones"][0].__dict__["pods"][0].__dict__["clusters"][0].__dict__["hosts"][0].__dict__
```
Comment (nvazquez) [inline test/integration/component/test_host_ha.py]: Same here

File: test/integration/component/test_overcommit.py [hunk 4]
```diff
@@ -99,8 +100,9 @@ def setUpClass(cls):
         cls.domain = get_domain(cls.apiclient)
         cls.zone = get_zone(cls.apiclient)
         cls.testdata["mode"] = cls.zone.networktype
-        cls.testdata["configurableData"]["password"] = "xenroot"
+        #cls.testdata["configurableData"]["password"] = "xenroot"
         cls.hypervisor = testClient.getHypervisorInfo()
+        cls.hostConfig = cls.config.__dict__["zones"][0].__dict__["pods"][0].__dict__["clusters"][0].__dict__["hosts"][0].__dict__
```
Comment (nvazquez) [inline test/integration/component/test_overcommit.py]: Same here

File: test/integration/component/test_overcommit.py [hunk 5]
```diff
@@ -99,8 +100,9 @@ def setUpClass(cls):
         cls.domain = get_domain(cls.apiclient)
         cls.zone = get_zone(cls.apiclient)
         cls.testdata["mode"] = cls.zone.networktype
-        cls.testdata["configurableData"]["password"] = "xenroot"
+        #cls.testdata["configurableData"]["password"] = "xenroot"
```
Comment (nvazquez) [inline test/integration/component/test_overcommit.py]: Can we remove this commented line?

File: test/integration/component/test_volumes.py [hunk 6]
```diff
@@ -123,17 +67,17 @@ def setUpClass(cls):
...
+        cls.template = get_template(
             cls.api_client,
             cls.zone.id,
-            cls.services["ostype"]
+            cls.testdata["ostype"]
         )
-        cls.services["zoneid"] = cls.zone.id
-        cls.services["virtual_machine"]["zoneid"] = cls.zone.id
-        cls.services["virtual_machine"]["template"] = template.id
+        #cls.testdata["zoneid"] = cls.zone.id
```
Comment (nvazquez) [inline test/integration/component/test_volumes.py]: Multiple commented lines

File: test/integration/component/test_VirtualRouter_alerts.py (hunk 1 above)
Comment (nvazquez) [reply test/integration/component/test_VirtualRouter_alerts.py]: Ok, sorry, copy that. Please ignore similar comments, thanks

**Comment by borisstoyanov** [comment]
Thanks @nvazquez, @rhtyd I think we're in a good position to merge now, lets address the new failures in separate PRs.

**Comment by rhtyd** [comment]
Thanks @borisstoyanov I'll merge this, as we'll deal with fixing rest of the failures in other PRs.

Merging this based on improved component test results and 2lgtms code reviews.

**Pull request merged**